src/main/kotlin/org/opensearch/commons/alerting/action/IndexWorkflowRequest.kt

```diff
@@ -0,0 +1,137 @@
+package org.opensearch.commons.alerting.action
+
+import org.opensearch.action.ActionRequest
+import org.opensearch.action.ActionRequestValidationException
+import org.opensearch.action.ValidateActions
+import org.opensearch.action.support.WriteRequest
+import org.opensearch.common.io.stream.StreamInput
+import org.opensearch.common.io.stream.StreamOutput
+import org.opensearch.commons.alerting.model.CompositeInput
+import org.opensearch.commons.alerting.model.Workflow
+import org.opensearch.rest.RestRequest
+import java.io.IOException
+import java.util.stream.Collectors
+
+class IndexWorkflowRequest : ActionRequest {
+    val workflowId: String
+    val seqNo: Long
+    val primaryTerm: Long
+    val refreshPolicy: WriteRequest.RefreshPolicy
+    val method: RestRequest.Method
+    var workflow: Workflow
+    val rbacRoles: List<String>?
+
+    private val MAX_DELEGATE_SIZE = 25
+
+    constructor(
+        workflowId: String,
+        seqNo: Long,
+        primaryTerm: Long,
+        refreshPolicy: WriteRequest.RefreshPolicy,
+        method: RestRequest.Method,
+        workflow: Workflow,
+        rbacRoles: List<String>? = null
+    ) : super() {
+        this.workflowId = workflowId
+        this.seqNo = seqNo
+        this.primaryTerm = primaryTerm
+        this.refreshPolicy = refreshPolicy
+        this.method = method
+        this.workflow = workflow
+        this.rbacRoles = rbacRoles
+    }
+
+    @Throws(IOException::class)
+    constructor(sin: StreamInput) : this(
+        workflowId = sin.readString(),
+        seqNo = sin.readLong(),
+        primaryTerm = sin.readLong(),
+        refreshPolicy = WriteRequest.RefreshPolicy.readFrom(sin),
+        method = sin.readEnum(RestRequest.Method::class.java),
+        workflow = Workflow.readFrom(sin) as Workflow,
+        rbacRoles = sin.readOptionalStringList()
+    )
+
+    override fun validate(): ActionRequestValidationException? {
+        var validationException: ActionRequestValidationException? = null
+
+        if (workflow.inputs.isEmpty()) {
+            validationException = ValidateActions.addValidationError(
+                "Input list can not be empty.", validationException
+            )
+            return validationException
+        }
+        if (workflow.inputs.size > 1) {
+            validationException = ValidateActions.addValidationError(
```

Comment:
Looks like some of these flows are not covered by the unit test. Consider looking into ``Codecov`` comments.

Reply:
> I see three different names being used here in the PR - Workflows, ChainedMonitor and Composite. Wondering if we need to arrive at the consistency in naming, especially when UX and Rest APIs should be using Composite Monitors as the name.

**Workflow** refers to the the new type of scheduled job.
**Composite** is the input type - it accepts a sequence of monitors to be executed as input.
**Chained Monitor** refers to chained monitor findings i.e. findings of one monitor are used as the source data for the next monitor which is 'chained' to it.